src/docfx/lib/JsonUtility.cs

```diff
@@ -125,6 +123,101 @@ public static JObject Merge(JObject first, IEnumerable<JObject> objs)
+            token.Traverse(errors, mappings, nullNodes);
+            foreach (var node in nullNodes)
+            {
+                node.Remove();
+            }
+            return (errors, token);
+        }
+
+        private static bool IsNullOrUndefined(this JToken token)
+        {
+            return
+                (token == null) ||
+                (token.Type == JTokenType.Null) ||
+                (token.Type == JTokenType.Undefined);
+        }
+
+        /// <summary>
+        /// Parse a string to JToken.
+        /// Validate null value during the process.
+        /// </summary>
+        private static (List<Error>, JToken) Deserialize(TextReader reader)
+        {
+            try
+            {
+                using (JsonReader json = new JsonTextReader(reader))
+                {
+                    return DefaultDeserializer.Deserialize<JToken>(json).ValidateNullValue();
+                }
+            }
+            catch (Exception ex)
+            {
+                var errors = new List<Error>
+                {
+                    Errors.JsonSyntaxError(ex),
+                };
+                return (errors, JValue.CreateNull());
+            }
+        }
+
+        private static void Traverse(this JToken token, List<Error> errors, JTokenSourceMap mappings, List<JToken> nullNodes)
+        {
+            if (token is JArray array)
+            {
+                foreach (var item in token.Children())
+                {
+                    if (item.IsNullOrUndefined())
+                    {
+                        LogWarningForNullValue(array, errors, mappings);
+                        nullNodes.Add(item);
+                    }
+                    else
+                    {
+                        Traverse(item, errors, mappings, nullNodes);
+                    }
+                }
+            }
+            else if (token is JObject obj)
+            {
+                foreach (var item in token.Children())
+                {
+                    var prop = item as JProperty;
+                    if (prop.Value.IsNullOrUndefined())
+                    {
+                        LogWarningForNullValue(token, errors, mappings);
+                        nullNodes.Add(item);
+                    }
+                    else
+                    {
+                        prop.Value.Traverse(errors, mappings, nullNodes);
+                    }
+                }
+            }
+        }
+
+        private static void LogWarningForNullValue(JToken item, List<Error> errors, JTokenSourceMap mappings)
+        {
+            if (mappings == null)
+            {
+                var lineInfo = item as IJsonLineInfo;
+                errors.Add(Errors.NullValue(new Range(lineInfo.LineNumber, lineInfo.LinePosition)));
+            }
+            else
+            {
+                Debug.Assert(mappings.ContainsKey(item));
+                if (mappings.TryGetValue(item, out Range value))
```

Comment:
nit: Given the assertion above, use `var value = mappings[item]` to throw an exception if something goes terribly wrong,